src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs

```diff
@@ -443,6 +443,71 @@ int F(int a)
                 Row(4, TableIndex.MethodDef, EditAndContinueOperation.Default));
         }
 
+        [ConditionalFact(AlwaysSkip = "")]
+        public void MethodWithNullable_AddingNullCheck()
+        {
+            var source0 = MarkedSource(@"
+using System;
+#nullable enable
+
+class C
+{
+    static T id<T>(T t) => t;
+    static T G<T>(Func<T> f) => f();
+
+    public void F(string? x)
+    <N:0>{</N:0>
+        var y1 = new { A = id(x) };
+        var y2 = G(() => new { B = id(x) });
+        var z = new Func<string>(() => y1.A + y2.B);
+    }
+}", options: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+            var source1 = MarkedSource(@"
+using System;
+#nullable enable
+
+class C
+{
+    static T id<T>(T t) => t;
+    static T G<T>(Func<T> f) => f();
+
+    public void F(string? x)
+    <N:0>{</N:0>
+        if (x is null) throw new Exception();
+        var y1 = new { A = id(x) };
+        var y2 = G(() => new { B = id(x) });
+        var z = new Func<string>(() => y1.A + y2.B);
```

Comment:
You need to mark all variable declaration nodes. See CaptureStructAndThroughClassEnvChain.